src/sign_dlink_ru.c

```diff
@@ -183,15 +187,27 @@ FILE* make_out_file(char* filename) {
  * Sign the firmware file after all of our checks have completed
  */
 void sign_firmware(char* filename, char** salts, int num_salts) {
-	int i;
+	int i = 0;
 	salt_t salt;
 	FILE* f = make_out_file(filename);
+	
+	char* version_suffix = "c0ffeef0rge";
```

Comment:
Instead of having to add an optional -v <suffix> argument (and a default when not specified), the code is simpler if the version suffix a mandatory argument and is specified when signing the firmware for the DIR-882 R1.

As a side note - If I remember correctly, the DIR-882 R1 doesn't actually care what this added suffix is. c0ffeef0rge was a random placeholder inspired by d-link's use of `0x00c0ffee`. This should be tested though

Reply:
as these are different pull requests I want to make it backward comparable
as some time it can take months to get a pull request through 
I didn't want it broken in between

starting out it did change the failed message from maybe bad model or magic to bad signature
but I have to gone back to check so worth doing}

PS I don't have this device others in forum have would make life easier if i did